hazelcast/src/test/java/com/hazelcast/map/MapIndexLifecycleTest.java

```diff
@@ -0,0 +1,195 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.MapIndexConfig;
+import com.hazelcast.config.MapStoreConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.core.MapLoader;
+import com.hazelcast.instance.Node;
+import com.hazelcast.map.impl.MapService;
+import com.hazelcast.map.impl.MapServiceContext;
+import com.hazelcast.map.impl.PartitionContainer;
+import com.hazelcast.map.impl.recordstore.RecordStore;
+import com.hazelcast.query.impl.Indexes;
+import com.hazelcast.spi.impl.NodeEngineImpl;
+import com.hazelcast.spi.properties.GroupProperty;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static com.hazelcast.config.InMemoryFormat.NATIVE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class MapIndexLifecycleTest extends HazelcastTestSupport {
+
+    private static final int BOOK_COUNT = 1000;
+
+    @Test
+    public void backupsShouldNotBeIndexedWhenThereIsNoMigration() {
+        TestHazelcastInstanceFactory instanceFactory = createHazelcastInstanceFactory(1);
+        HazelcastInstance instance1 = createNode(instanceFactory);
+
+        IMap bookMap = instance1.getMap("default");
+        assertEquals(BOOK_COUNT, bookMap.size());
+
+        bookMap.destroy();
+        assertAllPartitionContainersAreEmpty(instance1);
+
+        bookMap = instance1.getMap("default");
+        assertEquals(BOOK_COUNT, bookMap.size());
+
+        assertAllPartitionContainersAreInitialized(instance1);
+    }
+
+    private void assertAllPartitionContainersAreEmpty(HazelcastInstance instance) {
+        MapServiceContext context = getMapServiceContext(instance);
+        int partitionCount = getPartitionCount(instance);
+
+        for (int i = 0; i < partitionCount; i++) {
+            PartitionContainer container = context.getPartitionContainer(i);
+
+            ConcurrentMap<String, RecordStore> maps = container.getMaps();
+            assertTrue(maps.isEmpty());
```

Comment:
It would be great to add a message here, otherwise we just get a vacuous `AssertionError` on a failure.
```java
assertTrue("Expected empty RecordStores map in PartitionContainer, but found " + maps.size(), maps,isEmpty());
```